MathLib/LinAlg/UnifiedMatrixSetters.cpp

```diff
@@ -97,6 +103,12 @@ void setVector(PETScVector& v,
     v.set(idcs, vals);
 }
 
+void setVector(PETScVector& v, MatrixVectorTraits<PETScVector>::Index const index,
+               double const value)
+{
+    v.set(index, value);
```

Comment:
TODO: handle negative indices specially; and look for negative indices in PR #1130.